File: Backend/src/storage/seed.ts
import type { StoreState } from '../domain/types.js';

export const seedState: StoreState = {
  inspections: [],
  stations: [],
  alerts: [],
  users: [
    { id: 'u1', username: 'operator1', password: 'pass', name: 'Budi Santoso', role: 'operator' },
    { id: 'u2', username: 'qc1', password: 'pass', name: 'Siti Rahayu', role: 'qc' },
    { id: 'u3', username: 'supervisor1', password: 'pass', name: 'Ahmad Wijaya', role: 'supervisor' },
    { id: 'u4', username: 'eng1', password: 'pass', name: 'Dian Pratama', role: 'engineering' },
    { id: 'u5', username: 'admin', password: 'admin', name: 'Admin System', role: 'admin' },
    { id: 'u6', username: 'vendor1', password: 'pass', name: 'PT. Mitra Sejahtera', role: 'vendor' },
  ],
  parts: [
    {
      id: 'pt1',
      partName: 'Spur Gear A',
      partCode: 'SG-001',
      vendor: 'PT. Mitra Sejahtera',
      dimensions: [
        { id: 'd1', name: 'Outer Diameter', nominal: 12, upperLimit: 12.05, lowerLimit: 11.95, unit: 'mm' },
        { id: 'd2', name: 'Inner Diameter', nominal: 4, upperLimit: 4.03, lowerLimit: 3.97, unit: 'mm' },
        { id: 'd3', name: 'Thickness', nominal: 2.5, upperLimit: 2.55, lowerLimit: 2.45, unit: 'mm' },
      ],
    },
    {
      id: 'pt2',
      partName: 'Holder B',
      partCode: 'HLD-002',
      vendor: 'PT. Komponen Utama',
      dimensions: [
        { id: 'd4', name: 'Length', nominal: 25, upperLimit: 25.1, lowerLimit: 24.9, unit: 'mm' },
        { id: 'd5', name: 'Width', nominal: 15, upperLimit: 15.08, lowerLimit: 14.92, unit: 'mm' },
        { id: 'd6', name: 'Hole Diameter', nominal: 5, upperLimit: 5.05, lowerLimit: 4.95, unit: 'mm' },
      ],
    },
    {
      id: 'pt3',
      partName: 'Screw M3',
      partCode: 'SCR-003',
      vendor: 'PT. Baut Nusantara',
      dimensions: [
        { id: 'd7', name: 'Shaft Diameter', nominal: 3, upperLimit: 3.02, lowerLimit: 2.98, unit: 'mm' },
        { id: 'd8', name: 'Head Diameter', nominal: 5.5, upperLimit: 5.55, lowerLimit: 5.45, unit: 'mm' },
        { id: 'd9', name: 'Total Length', nominal: 8, upperLimit: 8.1, lowerLimit: 7.9, unit: 'mm' },
      ],
    },
    {
      id: 'pt4',
      partName: 'Pin Shaft C',
      partCode: 'PIN-004',
      vendor: 'PT. Mitra Sejahtera',
      dimensions: [
        { id: 'd10', name: 'Diameter', nominal: 2, upperLimit: 2.02, lowerLimit: 1.98, unit: 'mm' },
        { id: 'd11', name: 'Length', nominal: 10, upperLimit: 10.05, lowerLimit: 9.95, unit: 'mm' },
      ],
    },
  ],
  qualityRecords: [],
};

File: backend/src/storage/seed.rs
use crate::domain::*;

pub struct SeedUser {
    pub id: &'static str,
    pub username: &'static str,
    pub password: &'static str,
    pub name: &'static str,
    pub role: UserRole,
}

pub struct SeedPart {
    pub id: &'static str,
    pub part_name: &'static str,
    pub part_code: &'static str,
    pub vendor: &'static str,
    pub dimensions: &'static [SeedDimension],
}

pub struct SeedDimension {
    pub id: &'static str,
    pub name: &'static str,
    pub kind: DimensionKind,
    pub view: DimensionView,
    pub nominal: f64,
    pub upper_limit: f64,
    pub lower_limit: f64,
    pub unit: &'static str,
}

pub const USERS: &[SeedUser] = &[
    SeedUser { id: "u-001", username: "admin", password: "admin123", name: "Administrator", role: UserRole::Admin },
    SeedUser { id: "u-002", username: "supervisor", password: "super123", name: "Budi Santoso", role: UserRole::Supervisor },
    SeedUser { id: "u-003", username: "qc1", password: "qc123", name: "Sari Dewi", role: UserRole::Qc },
    SeedUser { id: "u-004", username: "operator1", password: "op123", name: "Andi Pratama", role: UserRole::Operator },
    SeedUser { id: "u-005", username: "vendor1", password: "ven123", name: "PT. Maju Jaya", role: UserRole::Vendor },
    SeedUser { id: "u-006", username: "engineer1", password: "eng123", name: "Rina Engineering", role: UserRole::Engineering },
];

const RDB_DIMENSIONS: &[SeedDimension] = &[
    SeedDimension {
        id: "d-001",
        name: "Diameter",
        kind: DimensionKind::Diameter,
        view: DimensionView::Top,
        nominal: 280.0,
        upper_limit: 280.5,
        lower_limit: 279.5,
        unit: "mm",
    },
    SeedDimension {
        id: "d-002",
        name: "Thickness",
        kind: DimensionKind::Width,
        view: DimensionView::Side,
        nominal: 22.0,
        upper_limit: 22.3,
        lower_limit: 21.7,
        unit: "mm",
    },
];

const BPS_DIMENSIONS: &[SeedDimension] = &[
    SeedDimension {
        id: "d-003",
        name: "Width",
        kind: DimensionKind::Width,
        view: DimensionView::Top,
        nominal: 55.0,
        upper_limit: 55.4,
        lower_limit: 54.6,
        unit: "mm",
    },
    SeedDimension {
        id: "d-004",
        name: "Length",
        kind: DimensionKind::Length,
        view: DimensionView::Top,
        nominal: 42.0,
        upper_limit: 42.3,
        lower_limit: 41.7,
        unit: "mm",
    },
];

pub const PARTS: &[SeedPart] = &[
    SeedPart {
        id: "p-001",
        part_name: "Rotor Disc Brake",
        part_code: "RDB-001",
        vendor: "PT. Maju Jaya",
        dimensions: RDB_DIMENSIONS,
    },
    SeedPart {
        id: "p-002",
        part_name: "Brake Pad Set",
        part_code: "BPS-002",
        vendor: "PT. Maju Jaya",
        dimensions: BPS_DIMENSIONS,
    },
];

impl From<&SeedDimension> for DimensionSpec {
    fn from(value: &SeedDimension) -> Self {
        Self {
            id: value.id.to_string(),
            name: value.name.to_string(),
            kind: value.kind,
            view: value.view,
            nominal: value.nominal,
            upper_limit: value.upper_limit,
            lower_limit: value.lower_limit,
            unit: value.unit.to_string(),
        }
    }
}
